### PR TITLE
Bug 1163799: Use Celery soft timeouts rather than the validator's pro…

### PR DESCRIPTION
**Pull request by kmaglione**
…blematic setitimer/SIGALRM implementation.

### CODE REVIEW

**Comment by kmaglione** [comment]
This relies on mozilla/amo-validator#300, so we're going to need to update the amo-validator requirement before merging.

File: apps/devhub/tasks.py [hunk 1]
```diff
@@ -32,21 +32,23 @@
...
 def validator(upload_id, listed=True, **kw):
     if not settings.VALIDATE_ADDONS:
         return None
+
     log.info('VALIDATING: %s' % upload_id)
     upload = FileUpload.objects.using('default').get(pk=upload_id)
     try:
         result = run_validator(upload.path, listed=listed)
-    except:
+    except Exception:
```
Comment (magopian) [inline apps/devhub/tasks.py]: That's the same as just having an `except`, right?

**Comment by magopian** [comment]
That's a good idea, let's hope it works as expected ;)

As said above, do not merge before https://github.com/mozilla/amo-validator/pull/300 has landed and a new release is made (and then the requirements are updated in this PR).

File: apps/devhub/tasks.py (hunk 1 above)
Comment (kmaglione) [inline apps/devhub/tasks.py]: No. A bare except catches things which inherit from `BaseException`, but not `Exception`, which generally shouldn't be caught by an exception handler. This has become a common anti-pattern: http://stackoverflow.com/questions/18982610/difference-between-except-and-except-exception-as-e-in-python

Comment (magopian) [inline apps/devhub/tasks.py]: ah, so you explicitely don't want to catch the "system exiting" exceptions, I understand. Please add a comment to explain that, thanks!

**Pull request merged**